jdaviz/app.py

```diff
@@ -537,8 +548,14 @@ def _change_reference_data(self, new_refdata_label):
                 float(fov_sky_final / fov_sky_init)
             )
 
-        # re-center the viewer on previous location
-        viewer.center_on(sky_cen)
+        # only re-center the viewer if all data layers have WCS:
+        data_has_wcs = [
+            hasattr(d, 'coords') and isinstance(d.coords, BaseHighLevelWCS)
+            for d in viewer.data()
+        ]
```

Comment:
I can't remember - did we come up with a way to handle this otherwise?  Is that in scope here or as a follow-up?

Reply:
I don't think we settled on a complete solution.

Reply:
I suspect we might want to still have some recentering as _anything_ will feel more consistent to the user than just skipping the viewer, but we can revisit later if that isn't easy to implement here.